### PR TITLE
2_part

### PR DESCRIPTION
**Pull request by YaTut1901**

### CODE REVIEW

**Comment by dariiabutorina** [comment]
All suggestions are in dm

File: src/main/java/main/internet/shop/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public Order completeOrder(ShoppingCart cart) {
+        Order order = new Order(cart.getUserId());
+        order.getProducts().addAll(List.copyOf(cart.getProducts()));
+        cart.getProducts().clear();
+        return order;
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
```
Comment (Sofasmile) [inline src/main/java/main/internet/shop/service/impl/OrderServiceImpl.java:30]: let's create this method at the dao layer

File: src/main/java/main/internet/shop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public List<Order> getAll() {
+        return orderDao.getAll();
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return orderDao.delete(get(id));
```
Comment (Sofasmile) [inline src/main/java/main/internet/shop/service/impl/OrderServiceImpl.java]:
```suggestion
        return orderDao.delete(id);
```

File: src/main/java/main/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public void clear(ShoppingCart cart) {
+        cart.getProducts().clear();
+        shoppingCartDao.update(cart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getAll().stream()
```
Comment (Sofasmile) [inline src/main/java/main/internet/shop/service/impl/ShoppingCartServiceImpl.java]: let's move this method on the dao layer

File: src/main/java/main/internet/shop/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Override
+    public User update(User user) {
+        return userDao.update(user);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return userDao.delete(get(id));
```
Comment (Sofasmile) [inline src/main/java/main/internet/shop/service/impl/UserServiceImpl.java]:
```suggestion
        return userDao.delete(id);
```

File: src/main/java/main/internet/shop/dao/impl/OrderDaoImpl.java [hunk 5]
```diff
@@ -28,6 +29,13 @@ public Order create(Order order) {
         return Storage.orders;
     }
 
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return Storage.orders.stream()
+                .filter(order -> order.getUserId() == userId)
```
Comment (Sofasmile) [inline src/main/java/main/internet/shop/dao/impl/OrderDaoImpl.java]: == or equals()?

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**